lib/library.rb

```diff
@@ -0,0 +1,64 @@
+require 'yaml'
+require 'date'
+require './lib/person.rb'
+
+class Library
+  attr_accessor :books, :title, :available, :renter, :return_date
+
+  def initialize(books, title, available, renter, return_date)
+    @books = books
+    @title = title
+    @available = available
+    @renter = name
+    @return_date = date.today.next_month
+
+
+  end
+
+  def list_available
+    books.select { |obj| obj[:available] == true }
+  end
+
+  def list_all
+    YAML.load_file('./lib/books.yml')
+  end
+
+  def search_books
+    def search_books(title)
+  book = books.select { |obj| obj[:item][:title] == title}
+  if book[:available] == false
+    'Not available'
+  else
+    book
+  end
+end
+
+def book
+  books.select { |obj| obj[:available] == true}
+  @books[:available] = false
+  @books[:return_date]= @return_date
+  @books[:renter] = name
+  change_books
+end
+
+
+  def check_outdate
+    check = books.select{ |obj| obj[:item][:title].include? ''}
+    if check = [:available] == false
+      return @return_date
+    end
+  end
+
+  def set_outdate
+    Date.today.next_month
+  end
+
+  def books
+    YAML.load_file('./lib/books.yml')
```

Comment:
Aha, now I see. what seem to be a variable in some methods, is actually a method. It's a bit hard to follow due to the name. Naming methods are a tricky thing and that's a different discussion. In this case I would argue that you need to store books in `@books` attribute and not read them from the data file each time you want to access them.